library/oneview_server_profile.py

```diff
@@ -352,9 +387,10 @@ def __update_server_profile(self, profile_with_updates, original_profile):
             error_msg = '; '.join(str(e) for e in exception.args)
             power_on_msg = 'Some server profile attributes cannot be changed while the server hardware is powered on.'
             if power_on_msg in error_msg:
-                time.sleep(10)
                 logger.debug("Update failed due to powered on Server Hardware. Powering off before retrying.")
+                time.sleep(10)  # sleep timer to avoid timing issues after update operation failed
                 self.__set_server_hardware_power_state(original_profile['serverHardwareUri'], 'Off')
+                self.__set_server_hardware_power_state(profile_with_updates['serverHardwareUri'], 'Off')
```

Comment:
Why setting the power state to both?
Did you forget to remove anything here? If not please comment this part of the code.

Reply:
If we're reassigning server hardwares, both the original server hardware  and the new server hardware have to be off, so this checks for both and sets any to necessary to off. 
Lemme include a comment in the code to make this clear.